libs/feeds_processing/src/utils.rs

```diff
@@ -605,3 +370,161 @@ pub async fn validate(
+                end_slot_timestamp: 1677654321,
+            },
+            VotedFeedUpdate {
+                feed_id: 5,
+                value: FeedType::Numerical(aggregated_values[1]),
+                end_slot_timestamp: 1677654322,
+            },
+            VotedFeedUpdate {
+                feed_id: 11,
+                value: FeedType::Numerical(aggregated_values[2]),
+                end_slot_timestamp: 1677654323,
+            },
+        ];
+
+        let mut feeds_rounds: HashMap<u32, u64> = HashMap::new();
+        feeds_rounds.insert(1, 1000);
+        feeds_rounds.insert(5, 2000);
+        feeds_rounds.insert(11, 3000);
+        feeds_rounds.insert(3, 4000);
+
+        let block_height = 100;
+        let network = "ETH".to_string();
+
+        let updates_to_serialize = BatchedAggegratesToSend {
+            block_height,
+            updates: updates.clone(),
+            proofs: HashMap::new(),
+        };
+
+        let calldata = adfs_serialize_updates(
+            network.as_str(),
+            &updates_to_serialize,
+            None,
+            create_feeds_config(),
+            &mut feeds_rounds,
+        )
+        .await
+        .unwrap();
+
+        let consensus_second_rond_batch = ConsensusSecondRoundBatch {
+            sequencer_id: 0,
+            block_height,
+            network,
+            contract_address: "0x663F3ad617193148711d28f5334eE4Ed07016602".to_string(),
+            safe_address: "0x7f09E80DA1dFF8df7F1513E99a3458b228b9e19C".to_string(),
+            nonce: "10".to_string(),
+            chain_id: "31337".to_string(),
+            tx_hash: "0x1c856b6abec5d4168b8bdd0509da6f84a486081c19ba2e49e8acc28af6d615dc"
+                .to_string(),
+            calldata,
+            updates,
+            feeds_rounds,
+            proofs: HashMap::new(),
+        };
+
+        validate(
+            create_feeds_config(),
+            consensus_second_rond_batch,
+            last_votes,
+            HashMap::new(),
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn test_validate_with_confirmable_input() {
+        // All the last votes of the reporter are within 1% deviation from the aggregated values.
+        match call_validate_with_values([42.1, 110.6, 552.0], [42.5, 111.5, 555.5]).await {
+            Ok(_) => {}
+            Err(e) => {
+                panic!("validate failed with error: {e}")
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_validate_with_non_confirmable_input() {
+        // The third vote of the reporter is not within 1% deviation from the aggregated value.
+        match call_validate_with_values([42.1, 110.6, 549.5], [42.5, 111.5, 555.5]).await {
+            Ok(_) => {
+                panic!("validate confirmation of higher than 1% deviation!");
+            }
+            Err(e) => {
+                info!("validate correctly did not approve batch of aggregates: {e}")
+            }
+        }
```

Comment:
Consider asserting that the error message contains the feed report that is outside the threshold.

Reply:
Addressed in [PR-1153](https://github.com/blocksense-network/blocksense/pull/1153)